src/Plugins/ComplexCore/src/ComplexCore/Filters/FindNumFeaturesFilter.cpp

```diff
@@ -0,0 +1,102 @@
+#include "complex/Filter/Actions/EmptyAction.hpp"
+#include "complex/Parameters/ArrayCreationParameter.hpp"
+#include "complex/Parameters/ArraySelectionParameter.hpp"
+#include "complex/Parameters/AttributeMatrixSelectionParameter.hpp"
+
+using namespace complex;
+
+namespace complex
+{
+//------------------------------------------------------------------------------
+std::string FindNumFeaturesFilter::name() const
+{
+  return FilterTraits<FindNumFeaturesFilter>::name.str();
+}
+
+//------------------------------------------------------------------------------
+std::string FindNumFeaturesFilter::className() const
+{
+  return FilterTraits<FindNumFeaturesFilter>::className;
+}
+
+//------------------------------------------------------------------------------
+Uuid FindNumFeaturesFilter::uuid() const
+{
+  return FilterTraits<FindNumFeaturesFilter>::uuid;
+}
+
+//------------------------------------------------------------------------------
+std::string FindNumFeaturesFilter::humanName() const
+{
+  return "Find Number of Features";
+}
+
+//------------------------------------------------------------------------------
+std::vector<std::string> FindNumFeaturesFilter::defaultTags() const
+{
+  return {"#Statistics", "#Morphological"};
+}
+
+//------------------------------------------------------------------------------
+Parameters FindNumFeaturesFilter::parameters() const
+{
+  Parameters params;
+
+  params.insertSeparator(Parameters::Separator{"Required Objects: Feature Data"});
+  params.insert(std::make_unique<ArraySelectionParameter>(k_FeaturePhasesArrayPath_Key, "Feature Phases", "", DataPath({"DataContainer", "FeatureData", "Phases"}), complex::GetAllDataTypes()));
+
+  params.insertSeparator(Parameters::Separator{"Created Objects: Ensemble Data"});
+  params.insert(std::make_unique<ArrayCreationParameter>(k_NumFeaturesArrayPath_Key, "Number of Features", "", DataPath({"Number of Features"})));
+
+  return params;
+}
+
+//------------------------------------------------------------------------------
+IFilter::UniquePointer FindNumFeaturesFilter::clone() const
+{
+  return std::make_unique<FindNumFeaturesFilter>();
+}
+
+//------------------------------------------------------------------------------
+IFilter::PreflightResult FindNumFeaturesFilter::preflightImpl(const DataStructure& dataStructure, const Arguments& filterArgs, const MessageHandler& messageHandler,
+                                                              const std::atomic_bool& shouldCancel) const
+{
+  auto pFeaturePhasesArrayPathValue = filterArgs.value<DataPath>(k_FeaturePhasesArrayPath_Key);
+  auto pNumFeaturesArrayPathValue = filterArgs.value<DataPath>(k_NumFeaturesArrayPath_Key);
+
+  PreflightResult preflightResult;
+  complex::Result<OutputActions> resultOutputActions;
+  std::vector<PreflightValue> preflightUpdatedValues;
+
+  const auto* featureData = dataStructure.getDataAs<AttributeMatrix>(pNumFeaturesArrayPathValue.getParent());
```

Comment:
@imikejackson Do we have a preferred way of specifying the parameters for cases like these?  I usually like to use the AttributeMatrixSelectionParameter along with the DataObjectNameParameter to achieve this since it seems more clear/less ambiguous to the user as to what is going to happen and its not really any more "clicks" for the user to enter it this way but just using an ArrayCreationParameter and checking the parent is an AttributeMatrix achieves the same thing in less parameters... thoughts?

Reply:
Less ambiguous meaning you can more clearly label what AttributeMatrix they are supposed to be selecting